mozdownload/treeherder.py

```diff
@@ -82,11 +82,11 @@ def query_builds_by_revision(self, revision, job_type_name='Build', debug_build=
 
             # Set filters to speed-up querying jobs
             kwargs = {
-                'platform': self.platform,
                 'option_collection_hash': option_hash,
                 'job_type_name': job_type_name,
                 'exclusion_profile': False,
             }
+            kwargs.update(self.get_treeherder_platform(self.platform))
```

Comment:
That would be a coding error but you are right. We should better check directly and not return the platform. Instead raise a NotSupportedError for it. I will include that in my next commit.